pandas/core/window.py

```diff
@@ -1044,18 +1055,26 @@ def validate(self):
                                           "for datetimelike and offset "
                                           "based windows")
 
+            if self.left_closed is not None and not is_bool(self.left_closed):
+                raise ValueError("left_closed must be a boolean")
+
             # this will raise ValueError on non-fixed freqs
             self.window = freq.nanos
+            if self.left_closed:
+                self.window += 1
```

Comment:
AFAIU, offset-based rolling windows relies on the conversion to nanoseconds in the line right above, so adding one not only works but is the optimal strategy &mdash; it includes the left endpoint and nothing else (since nanosecond is the smallest time resolution available).